### PR TITLE
In subtransaction_payment jbuilder, get `created` from the paymentable, not subtransaction_payment

### PR DESCRIPTION
**Pull request by awadyas**
**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

Closes https://github.com/CommitChange/tix/issues/4040

### CODE REVIEW

**Review by wwahammy — CHANGES_REQUESTED** [review]
Hmmm, this does fix the problem in a sense but it's not quite the right layer.

We'll need to be able to search and sort SubtransactionPayments by their `created` value. That means, for this PR, `SubtransactionPayment#created` to be set or updated when the `legacy_payment.date` is set or changed.

**Comment by awadyas** [comment]
> We'll need to be able to search and sort SubtransactionPayments by their created value. That means, for this PR, SubtransactionPayment#created to be set or updated when the legacy_payment.date is set or changed.

@wwahammy How does this sound as an approach: Adding a method on `Payment` to sync the `date` from the legacy payment to `SubtransactionPayment#created`, and putting the sync method in `create` and `update` in `PaymentsController`, conditionally, only if the payment has an associated subtransaction payment?

**Comment by awadyas** [comment]
So something like this

(in payments_controller.rb)
```
def create
  ...
  @payment.sync_date_to_subtransaction_payment if @payment.subtransaction_payment
end

def update
  ...
  @payment.sync_date_to_subtransaction_payment if @payment.subtransaction_payment
end

def sync_date_to_subtransaction_payment
  subtransaction_payment.created = date
end
```

Also: is there a way to only update `created` when the `date` specifically is updated? I think it would be cool / even better if we could do that

**Comment by wwahammy** [comment]
Unfortunately, there are ways in which payments can be updated and created that are not through an http request. For example, a recurring donation could run and a payment would be created for that but no http request would have occurred.

This will have to happen in InsertDonation, UpdateDonation, UpdateTickets, etc.

**Review by wwahammy — COMMENTED** [review]
Looks good. Is it possible to unit test the changes in InsertTickets too?

**Comment by awadyas** [comment]
> Is it possible to unit test the changes in InsertTickets too?

Yes, forgot about that lol :joy: I"m on it

File: spec/lib/insert/insert_tickets_spec.rb [hunk 1]
```diff
@@ -357,6 +357,13 @@ def success_expectations
           it 'increases TicketPurchase by 1' do
             expect { result }.to change {TicketPurchase.count}.by(1)
           end
+
+          it 'creates a subtransaction payment whose creation date matches the legacy payment\'s date' do
+            legacy_payment = result['payment']
+            subtransaction_payment = legacy_payment.subtransaction_payment
+
+            expect(subtransaction_payment.created).to eq(legacy_payment.created_at)
```
Comment (wwahammy) [inline spec/lib/insert/insert_tickets_spec.rb]: This should be the `legacy_payment.date`. On offsite payments, date will not be the same as the `created_at` time.

```suggestion
            expect(subtransaction_payment.created).to eq(legacy_payment.date)
```

**Review by wwahammy — CHANGES_REQUESTED** [review]
One quick tweak I think

**Review by wwahammy — APPROVED** [review]

**Pull request merged**